# Parameters for Traveling Salesman Problem
tsp:
  clustering: 'random'            # [STUDENTS TODO: implement better method than the default] Clustering method of viewpoints: [random (default), kmeans]
  distance_estimates: 'euclidean' # Method for evaluating cost of paths in TSP heuristic: [euclidean (default), astar, rrt, rrtstar]

# Parameters for path planning
path_planner:
  timeout: 10.0              # (s) timeout for single start-to-goal planning
  method: 'rrt'              # [STUDENTS TODO: implement better method than the default] Method for generating start-to-goal paths: [euclidean, astar, rrt (default), rrtstar]
  straighten_paths: false    # [STUDENTS TODO: implement better method than the default] Path straightening flag (default: false)

  # RRT and RRT* method parameters
  rrt:
    branch_size: 2.5 # (m)
    sampling:
      method: 'uniform' # [STUDENTS TODO: implement better method than the default] Method for sampling random points: [uniform (default), gaussian]
      gaussian:
        stddev_inflation: 0.2 # Inflation of standard deviation in all axes during gaussian sampling (m)
    star:
      neighborhood: 1.0 # (m)

  # A* method parameters
  astar:
    grid_resolution: 0.2 # (m)

# Parameters for path smoothing
path_smoothing:
  sampling_step: 0.1  # (m)
  lookahead_dist: 0.3 # (m)

# Parameters for sampling the UAVs' trajectories
trajectory_sampling:
  with_stops: true # [STUDENTS TODO: implement better method than the default] If true, UAVs will stop at each waypoint (default: true)

# Parameters for UAV-to-UAV collision avoidance
collision_avoidance:
  method: 'none' # [STUDENTS TODO: implement better method than the default] [none (default), delay_2nd_till_1st_UAV_finishes, delay_till_no_collisions_occur]
